base/server/src/main/java/org/dogtagpki/server/cli/SubsystemIdGeneratorUpdateCLI.java

```diff
@@ -0,0 +1,289 @@
+        LdapAuthInfo authInfo = getAuthInfo(passwordStore, connInfo, ldapConfig);
+
+        PKISocketConfig socketConfig = cs.getSocketConfig();
+
+        PKISocketFactory socketFactory = new PKISocketFactory();
+        socketFactory.setSecure(connInfo.getSecure());
+        if (authInfo.getAuthType() == LdapAuthInfo.LDAP_AUTHTYPE_SSLCLIENTAUTH) {
+            socketFactory.setClientCertNickname(authInfo.getClientCertNickname());
+        }
+        socketFactory.init(socketConfig);
+
+        DatabaseConfig dbConfig = cs.getDatabaseConfig();
+
+        if (generatorType.equals("cert")){
+            updateSerialNumberRangeGenerator(
+                    socketFactory,
+                    connInfo,
+                    authInfo,
+                    dbConfig,
+                    baseDN,
+                    generator,
+                    cs.getHostname(),
+                    getSecurePort(cs));
+            cs.commit(false);
+        } else if (generatorType.equals("request")) {
+            updateRequestNumberRangeGenerator(
+                    socketFactory,
+                    connInfo,
+                    authInfo,
+                    dbConfig,
+                    baseDN,
+                    generator);
+            cs.commit(false);
+        } else {
+            throw new EBaseException("Generator type " + generatorType + " not supported.");            
+        }
+    }
+
+    protected void updateSerialNumberRangeGenerator(PKISocketFactory socketFactory, LdapConnInfo connInfo,
+            LdapAuthInfo authInfo, DatabaseConfig dbConfig, String baseDN, IDGenerator newGenerator,
+            String hostName, String securePort) throws Exception {
+        
+        if (newGenerator == IDGenerator.RANDOM && idGenerator != IDGenerator.RANDOM) {
+            logger.debug("Remove serial ranges from configuration");
+            dbConfig.remove(DatabaseConfig.MIN_SERIAL_NUMBER);
+            dbConfig.remove(DatabaseConfig.MAX_SERIAL_NUMBER);
+            dbConfig.remove(DatabaseConfig.SERIAL_INCREMENT);
+            dbConfig.remove(DatabaseConfig.SERIAL_LOW_WATER_MARK);
+            dbConfig.remove(DatabaseConfig.SERIAL_CLONE_TRANSFER_NUMBER);
+            dbConfig.remove(DatabaseConfig.SERIAL_RANGE_DN);
+            return;
+        }
+        if (newGenerator == IDGenerator.LEGACY_2 && idGenerator == IDGenerator.LEGACY) {
+            logger.debug("Repository: Updating ranges entry to hex format");
+
+            LdapBoundConnection conn = new LdapBoundConnection(socketFactory, connInfo, authInfo);
+            try{
+                String rangeDN = dbConfig.getSerialRangeDN() + "," + baseDN;
+
+                String beginSerialNumber = dbConfig.getBeginSerialNumber();
+                dbConfig.setBeginSerialNumber("0x" + beginSerialNumber);
+                
+                String endSerialNumber = dbConfig.getEndSerialNumber();
+                LDAPEntry entrySerial = conn.read("cn=" + beginSerialNumber+"," + rangeDN);
```

Comment:
There's a possibility that there's no range object created yet, for example:
https://github.com/dogtagpki/pki/blob/master/.github/workflows/ca-sequential-test.yml#L532-L537

How should this case be handled?

Reply:
I'm also not sure the format is correct. IIUC the `cn` is the same as `beginRange` which is decimal.

Reply:
If the range entry is not present then the configuration value is used. I have updated the code to handle the exception.

Reply:
`cn` is decimal but the string match the hex in `begineRange` because the radix error.